### PR TITLE
Syncing with MLFlow and adding i18n support to OSS

### PR DESCRIPTION
**Pull request by sunishsheth2009**
## What changes are proposed in this pull request?
- [x] Syncing with MLFlow
- [x] Adding i18n support to OSS

## How is this patch tested?
- [x] Ran the server locally to make sure that it works as expected
- [x] Test runs

## Release Notes
[2021-06-14] 333f64ef4564  ML-15548 Adding support for dev lang (Sunish Sheth)
[2021-06-11] 6ad46c3588c0  ML-15750 - ML-15880 Moving metrics to show before parameters and tags and update startTime (Sunish Sheth)
[2021-06-08] c7b4158dbb8b  [ML-15486] Add i18n to Model Versions Page (Yun Park)
[2021-06-07] 585fd5c7459f  ML-15472 Adding i18n strings for labels in experiments view page (Sunish Sheth)
[2021-06-03] 54bd4f8af401  [ML-15485] Add i18n to Model Page (Yun Park)
[2021-06-02] b97f574bad0b  [ML-15484] Add i18n to Model List Page (Yun Park)
[2021-06-01] 2cca7e0f2fb9  ML-15472 Adding i18n strings for labels in experiments view page (Sunish Sheth)
[2021-05-28] 7773fd827ffc  ML-15472 Adding formatted messages for experiment view page (Sunish Sheth)
[2021-05-25] 7599e8965fff  ML-15470 Adding i18n scripts to MlFlow (Sunish Sheth)
[2021-05-24] 778905e5e752  [ML-13746] Add `creator` to model page UI (Steven Chen)
[2021-05-22] 9428974afc1c  (tag: master-58519-9428974a-Tag) [ML-14899] Permissions dialog and registry UI changes (Sue Ann Hong)
[2021-05-21] 41a20c59095a  (tag: master-58488-41a20c59-Tag) [ML-15450] Changed example UI limit to 5MB (Steven Chen)
[2021-05-18] 51ca84033e4d  [ML-14313] Add I18n infrastructure to MLflow (Alex Cheng)
[2021-05-03] 87457c181835  [ML-14808] Ensure job run is fetched even if state is already terminal (Sebastiaan Boer)
[2021-04-01] 66228bca0147  [ML-14635] Scroll to artifact viewer upon page load if initial artifact path is specified (Apurva Koti)
[2021-04-01] 96dd5eb10d61  [WARMFIX][ML-14764] [ML-14454] [ML-14453] Job failures are not surfaced on the Experiment page (Sebastiaan Boer)
[2021-03-29] 2ef658e7ce85  [ML-14039] Display tensor type in schema visualization (Arjun DCunha)
[2021-03-26] 1ce46a7cf21f  Expand API for Mlflow Button (Alex Cheng)
[2021-03-15] 24228213eb8d  [ML-14224] Save notebook IDs rather than URLs (Sebastiaan Boer)
[2021-03-09] 2f16a85803c3  [ML-13932] Include tags when downloading runs as csv (Harutaka Kawamura)
[2021-03-09] f4d0ac8b0997  [ML-14113] Allow users to access cluster settings tab at all times. (Arjun DCunha)
[2021-02-23] d844ca30492c  (tag: master-55927-d844ca30-Tag) [ML-13551] Add tag-based search to model registry UI (Juntai Zheng)
[2021-02-10] 6d20b2cb20d7  (tag: master-54614-6d20b2cb-Tag, tag: master-54613-6d20b2cb-Tag) [ML-13777] Create New Buttons for Experiments and Models (Mohamad Arabi)
[2021-02-06] 45cb787f8d71  [ML-13056] (redo) Elevate errors in MLflow UI to top.onerror (Apurva Koti)
[2021-02-03] 7b3c8d52a28c  [ML-13551] Add tags column to model registry UI (Juntai Zheng)
[2021-01-29] b59b2ce5675f  Use runs URI instead of dbfs for logged model code snippet (Apurva Koti)
[2021-01-15] 721c1f1be509  (tag: master-51623-721c1f1b-Tag) [ML-13419] [ES-57002] Update logic for choosing logged and registered model link to render (Apurva Koti)
[2021-01-13] 8082e3c7d958  (tag: master-51325-8082e3c7-Tag, tag: master-51324-8082e3c7-Tag) [ML-13348] Added copy button to ExperimentView page (Derrick Li)
[2021-01-04] 76a409aafd14  [ML-12414] Import OSS MLflow UI changes into Universe (Sid Murching)
[2020-12-10] e9661b183e88  Deduplicate logged models and fix flavor (Apurva Koti)
[2020-12-09] f5ed40be2c57  [ML-12598] URL encode model names when used in routes (Wendy Hu)
[2020-12-10] 545a4599ad8e  (tag: master-48890-545a4599-Tag) Deduplicate logged models and fix flavor (Apurva Koti)
[2020-11-20] 7d1ab48d4de1  [ML-12826] Rename Activity fields, CreateTranstionRequest return field name (Sue Ann Hong)
[2020-11-13] adbad60b9134  (tag: master-46676-adbad60b-Tag) [ML-12732] Add artifact view type + schema table for logged models (Apurva Koti)
[2020-11-12] 2d7855709c46  (tag: master-46514-2d785570-Tag) [ML-11300] Rename returned field of ListTransitionRequests from activities to requests (Sue Ann Hong)
[2020-10-15] 3d2f0527b6aa  (tag: branch-3.31-3148-3d2f0527-Tag) [WARMFIX] Revert ML-12244 (Jimmy Xu)
[2020-10-30] 41f30a0b2b92  (tag: master-45472-41f30a0b-Tag) [ML-12706] Experiment Runs Table will no longer nest when filtering or sorting (Jimmy Xu)
[2020-10-28] 07164a65591a  (tag: master-45238-07164a65-Tag, tag: master-45237-07164a65-Tag) [ML-12477] Add URL route for artifact preselection in run page (Apurva Koti)
[2020-10-23] 7f3e4b372185  [ML-12578] Removing "Preview" icon from main Models Page. (Mani Parkhe)
[2020-10-15] 84e97712b600  (tag: master-44189-84e97712-Tag, tag: master-44188-84e97712-Tag) [ML-12432] Replace activity.permission_level to available_actions (Andy Chow)
[2020-09-28] 5e0d0653a945  (tag: master-42713-5e0d0653-Tag, tag: master-42712-5e0d0653-Tag) [ML 11908] Implement PendingRequestsTableContainer (Jimmy Xu)
[2020-09-09] 16d914be6d77  [ML-11543] Improve ML Onboarding CUJs (Arjun DCunha)
[2020-08-24] c8904d33b589  (tag: master-39746-c8904d33-Tag) Set Mlflow Clone Run label & set up feature flag (Will Zhang)
[2020-08-17] 1c6cd38db493  (tag: master-39265-1c6cd38d-Tag) [ML-11552] Reflect MLflow run selection in notebook URL and vice versa (Apurva Koti)
[2020-08-06] 7fb0d114a550  [ES-32383] Fix Model Version Page crash when schema failed to parse (zhidongqu-db)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (xanderwebs) [inline mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx:1]: This folder should probably not be synced over to OS cc @smurching

**Review by xanderwebs — APPROVED** [review]
Read through the js/src/shared and the relevant i18n changes, lgtm so far. Would be good to get some eyes on the other files in this PR

Edit: the feedback link probably needs to be removed or added to edge.

File: mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx (hunk 1 above)
Comment (dbczumar) [reply mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx:1]: @sunishsheth2009 Let's drop this from the current PR (and any references to it) & file a follow-up task for Databricks to exclude this from OSS.

**Review by dbczumar — CHANGES_REQUESTED** [review]
@sunishsheth2009 For Tracking / Models, I tried out creating a run, setting run tags, and setting experiment notes, which all worked. For the Registry, I tried creating a model version, adding tags / description, and navigating around the Model Registry UI. Unfortunately, I found that the Registered Model Page does not load on the PR branch but loads successfully on `master`.

For example, I used the following code then navigated to http://localhost:3000/#/models/mymod, which yielded a blank page:

```
import mlflow

mlflow.set_tracking_uri("http://localhost:5000")

class Mod(mlflow.pyfunc.PythonModel):
    def predict(self, ctx, inp):
        return 7


with mlflow.start_run():
    mlflow.pyfunc.log_model(
        python_model=Mod(),
        artifact_path="mod",
        registered_model_name="mymod",
    )
```

Here's an error trace from chrome devtools:

```
index.js:1437 Warning: Each child in an array or iterator should have a unique "key" prop.

Check the render method of `ModelVersionViewImpl`. See https://fb.me/react-warning-keys for more information.
    in DescriptionsItem (at ModelVersionView.js:242)
    in ModelVersionViewImpl
    in injectIntl(ModelVersionViewImpl) (created by Connect(injectIntl(ModelVersionViewImpl)))
    in Connect(injectIntl(ModelVersionViewImpl)) (at ModelVersionPage.js:204)
    in RequestStateWrapper (created by Connect(RequestStateWrapper))
    in Connect(RequestStateWrapper) (at ModelVersionPage.js:184)
    in div (at ModelVersionPage.js:183)
    in ModelVersionPageImpl (created by Connect(ModelVersionPageImpl))
    in Connect(ModelVersionPageImpl)
    in Route (at App.js:98)
    in Switch (at App.js:88)
    in AppErrorBoundary (at App.js:87)
    in div (at App.js:41)
    in Router (created by HashRouter)
    in HashRouter (at App.js:40)
    in App (created by Connect(App))
    in Connect(App) (at index.js:21)
    in Provider (at index.js:20)
    in IntlProvider (at index.js:19)
console.<computed> @ index.js:1437
```

Fortunately, I confirmed that the Registered Model UI loads properly on Databricks, so this an OSS sync issue.

@sunishsheth2009 @xanderwebs Let me know if you can reproduce this and debug. Thanks for your help :)

**Comment by sunishsheth2009** [comment]
> @sunishsheth2009 For Tracking / Models, I tried out creating a run, setting run tags, and setting experiment notes, which all worked. For the Registry, I tried creating a model version, adding tags / description, and navigating around the Model Registry UI. Unfortunately, I found that the Registered Model Page does not load on the PR branch but loads successfully on `master`.
> 
> For example, I used the following code then navigated to http://localhost:3000/#/models/mymod, which yielded a blank page:
> 
> ```
> import mlflow
> 
> mlflow.set_tracking_uri("http://localhost:5000")
> 
> class Mod(mlflow.pyfunc.PythonModel):
>     def predict(self, ctx, inp):
>         return 7
> 
> 
> with mlflow.start_run():
>     mlflow.pyfunc.log_model(
>         python_model=Mod(),
>         artifact_path="mod",
>         registered_model_name="mymod",
>     )
> ```
> 
> Here's an error trace from chrome devtools:
> 
> ```
> index.js:1437 Warning: Each child in an array or iterator should have a unique "key" prop.
> 
> Check the render method of `ModelVersionViewImpl`. See https://fb.me/react-warning-keys for more information.
>     in DescriptionsItem (at ModelVersionView.js:242)
>     in ModelVersionViewImpl
>     in injectIntl(ModelVersionViewImpl) (created by Connect(injectIntl(ModelVersionViewImpl)))
>     in Connect(injectIntl(ModelVersionViewImpl)) (at ModelVersionPage.js:204)
>     in RequestStateWrapper (created by Connect(RequestStateWrapper))
>     in Connect(RequestStateWrapper) (at ModelVersionPage.js:184)
>     in div (at ModelVersionPage.js:183)
>     in ModelVersionPageImpl (created by Connect(ModelVersionPageImpl))
>     in Connect(ModelVersionPageImpl)
>     in Route (at App.js:98)
>     in Switch (at App.js:88)
>     in AppErrorBoundary (at App.js:87)
>     in div (at App.js:41)
>     in Router (created by HashRouter)
>     in HashRouter (at App.js:40)
>     in App (created by Connect(App))
>     in Connect(App) (at index.js:21)
>     in Provider (at index.js:20)
>     in IntlProvider (at index.js:19)
> console.<computed> @ index.js:1437
> ```
> 
> Fortunately, I confirmed that the Registered Model UI loads properly on Databricks, so this an OSS sync issue.
> 
> @sunishsheth2009 @xanderwebs Let me know if you can reproduce this and debug. Thanks for your help :)

image

Works fine now. Fixed the issue. Thanks for catching it.

File: mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx (hunk 1 above)
Comment (sunishsheth2009) [reply mlflow/server/js/src/shared/databricks_edge/CloudConditionalFeedbackLink.jsx]: Done!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @sunishsheth2009 !

**Pull request merged**